templates/guides/register-external-webhooks.md

```diff
@@ -0,0 +1,148 @@
+---
+page_title: "Register External Webhooks"
+description: "Register External Webhooks with Hookdeck Provider"
+---
+
+# Register External Webhooks
+
+Hookdeck Terraform provider provides a solution to help register and unregister webhooks with your providers. It's provider-agnostic, so as long as there's an endpoint to manage webhooks programmatically, you should be able to configure it as part of your workflow.
+
+## Setup
+
+For example, let's say you're using Stripe for payment and want to listen to `charge.succeeded` and `charge.failed` webhook events. Let's start with connection to start listening to incoming webhooks from Stripe.
+
+```hcl
+resource "hookdeck_source" "stripe" {
+  name = "stripe"
+}
+
+resource "hookdeck_destination" "payment_service" {
+  name = "my_destination"
+  url  = "https://api.my-app.com/webhooks/stripe"
+}
+
+resource "hookdeck_connection" "stripe_payment_service" {
+  source_id      = hookdeck_source.stripe.id
+  destination_id = hookdeck_destination.payment_service.id
+}
+```
+
+## Register Stripe webhook
+
+Stripe provides API endpoints to [create a new webhook](https://stripe.com/docs/api/webhook_endpoints/create) and [delete existing webhook](https://stripe.com/docs/api/webhook_endpoints/delete) which we will use in this example.
+
+```hcl
+resource "webhook_registration" "stripe" {
+  provider = hookdeck
+
+  register = {
+    request = {
+      method = "POST"
+      url    = "https://api.stripe.com/v1/webhook_endpoints"
+      headers = jsonencode({
+        "content-type" = "application/json"
+        authorization  = "Bearer <STRIPE_SECRET_KEY>"
+      })
+      body = jsonencode({
+        url = hookdeck_source.stripe.url
+        enabled_events = [
+          "charge.failed",
+          "charge.succeeded"
+        ]
+      })
+    }
+  }
+  unregister = {
+    request = {
+      method = "DELETE"
+      url    = "https://api.stripe.com/v1/webhook_endpoints/{{.register.response.body.id}}"
+      headers = jsonencode({
+        authorization  = "Bearer <STRIPE_SECRET_KEY>"
+      })
+    }
+  }
+}
+```
+
+For many API, you will need the ID of the registered webhook to unregister. You can usually get the ID of the newly registered webhook from the register API response itself. Because of that, Hookdeck Terraform provider will save that response for use in the `unregister` flow. The string values when you configure your `unregister` property are string templates where you'll have access to `.register.response` data which is the HTTP response from the original registration API request.
```

Comment:
```suggestion
For many APIs, you will need the ID of the registered webhook to unregister. You can usually get the ID of the newly registered webhook from the register API response itself. Because of that, the Hookdeck Terraform provider will save that response for use in the `unregister` flow. When configuring your `unregister` property, the string values are string templates where you'll have access to `.register.response` data, which is the HTTP response from the original registration API request.
```